hazelcast/src/main/java/com/hazelcast/spi/merge/HyperLogLogMergePolicy.java

```diff
@@ -29,22 +28,22 @@
  *
  * @since 3.10
  */
-public class HyperLogLogMergePolicy extends AbstractMergePolicy {
+public class HyperLogLogMergePolicy extends AbstractSplitBrainMergePolicy {
 
     public HyperLogLogMergePolicy() {
     }
 
     @Override
-    public <K, V> V merge(SplitBrainMergeEntryView<K, V> mergingEntry, SplitBrainMergeEntryView<K, V> existingEntry) {
-        if (!(mergingEntry.getValue() instanceof HyperLogLog)) {
-            throw new IllegalArgumentException("Unsupported merging entries");
+    public <V> V merge(MergeDataHolder<V> mergingData, MergeDataHolder<V> existingData) {
+        if (!(mergingData.getValue() instanceof HyperLogLog)) {
```

Comment:
is this `mergingData.getValue()` returning a `Data` type? or deserialized value type?

Reply:
It's returning whatever the data structure provides. So for `IMap` this can be `Data` or `Object`, for some data structures it will always be `Data` (I think collections have no in-memory format) and for the `CardinalityEstimator` it may always be `HyperLogLog`.

Reply:
This is the JavaDoc of `MergeDataHolder`, I hope it's clear enough:
```java
    /**
     * Returns the merging value in the in-memory format of the backing data structure.
     *
     * @return the merging value
     */
    V getValue();

    /**
     * Returns the deserialized merging value.
     *
     * @return the deserialized merging value
     */
    Object getDeserializedValue();
```

Reply:
yes it seems fragile, if internal impls' storage type changes, converting it to object seems can be safer